shared/src/validation-schemas/sign-up/sign-up.schema.ts

```diff
@@ -0,0 +1,15 @@
+import * as Joi from 'joi';
+import { UserCreatePayload } from '~/common/types/types';
+
+const SignupSchema = Joi.object<UserCreatePayload>({
+  firstName: Joi.string().min(1).max(20).required(),
```

Comment:
```suggestion
  [CreateUserPayload.FIRST_NAME]: Joi.string().min(1).max(20).required(),
```
use enum keys here****